orte/mca/oob/tcp/oob_tcp_listener.c

```diff
@@ -748,18 +741,18 @@ static void* listen_thread(opal_object_t *obj)
                                                 &addrlen);
                 if (pending_connection->fd < 0) {
                     OBJ_RELEASE(pending_connection);
-                    if (opal_socket_errno != EAGAIN || 
-                        opal_socket_errno != EWOULDBLOCK) {
-                        if (EMFILE == opal_socket_errno) {
-                            ORTE_ERROR_LOG(ORTE_ERR_SYS_LIMITS_SOCKETS);
-                            orte_show_help("help-orterun.txt", "orterun:sys-limit-sockets", true);
-                        } else {
-                            opal_output(0, "mca_oob_tcp_accept: accept() failed: %s (%d).",
-                                        strerror(opal_socket_errno), opal_socket_errno);
-                        }
+                    if (EAGAIN == opal_socket_errno ||
+                        EWOULDBLOCK == opal_socket_errno) {
+                        continue;
+                    } else if (EMFILE == opal_socket_errno) {
+                        ORTE_ERROR_LOG(ORTE_ERR_SYS_LIMITS_SOCKETS);
+                        orte_show_help("help-orterun.txt", "orterun:sys-limit-sockets", true);
+                        goto done;
+                    } else {
+                        opal_output(0, "mca_oob_tcp_accept: accept() failed: %s (%d).",
+                                    strerror(opal_socket_errno), opal_socket_errno);
                         goto done;
                     }
-                    continue;
```

Comment:
The behavior is actually the same.  I changed the logic flow a bit to make it a bit more clear (IMHO), but if you don't like my changes, I can put it back.

Here's the original logic: https://github.com/open-mpi/ompi/blob/master/orte/mca/oob/tcp/oob_tcp_listener.c#L749-L763

The old and new behavior is:
* If we get EAGAIN/EWOULDBLOCK, continue.
* If we get EMFILE, print a show_help and quit the listener
* If we get any other error, opal_output (which is kinda crummy... perhaps I should fix that) and quit the listener